## Github Repository for
# Ecological selection for small microbial genomes along a temperate-to-thermal soil gradient
## by Jackson W. Sorensen, Taylor K. Dunivin, Tammy C. Tobin & Ashley Shade

<i>This work is published.</i>


### Data
Metagenome data are available on IMG under the GOLD Study ID GS0114513. MG-RAST data are available under Project IDs mgp3731, mgp252, mgp5588, mgp14596, mgp6377, mgp6368, mgp2592, mgp2076, mgp11628, mgp13948, mgp7176 and mgp15600.


### To cite this work or code
Sorensen, J.W., Dunivin, T.K., Tobin, T.C. et al. Ecological selection for small microbial genomes along a temperate-to-thermal soil gradient. Nat Microbiol 4, 55–61 (2019). https://doi.org/10.1038/s41564-018-0276-6


### Abstract
Small bacterial and archaeal genomes provide insights into the minimal requirements for life1 and are phylogenetically widespread2. However, the precise environmental pressures that constrain genome size in free-living microorganisms are unknown. A study including isolates has shown that thermophiles and other bacteria with high optimum growth temperatures often have small genomes3. It is unclear whether this relationship extends generally to microorganisms in nature4,5 and more specifically to microorganisms that inhabit complex and highly variable environments, such as soils3,6,7. To understand the genomic traits of thermally adapted microorganisms, here we investigated metagenomes from a 45 °C gradient of temperate-to-thermal soils that lie over the ongoing Centralia, Pennsylvania (USA) coal-seam fire. We found that hot soils harboured distinct communities with small genomes and small cell sizes relative to those in ambient soils. Hot soils notably lacked genes that encode known two-component regulatory systems, and antimicrobial production and resistance genes. Our results provide field evidence for the inverse relationship between microbial genome size and temperature in a diverse, free-living community over a wide range of temperatures that support microbial life.

### Funding
This research was supported by Michigan State University and the National Science Foundation grant no. DEB1749544. Computational resources were provided by the Michigan State Institute for Cyber-Enabled Research. Metagenome sequencing was supported by the Joint Genome Institute Community Science Project no. 1834. The work conducted by the US Department of Energy Joint Genome Institute, a DOE Office of Science User Facility, is supported under Contract no. DE-AC02-05CH11231.


### More info
[ShadeLab](http://ashley17061.wixsite.com/shadelab/home)
